source/extensions/stat_sinks/common/statsd/statsd.h

```diff
@@ -29,15 +30,15 @@ class Writer : public ThreadLocal::ThreadLocalObject {
 public:
   Writer(Network::Address::InstanceConstSharedPtr address);
   // For testing.
-  Writer() : fd_(-1) {}
+  Writer() : io_handle_(std::make_unique<Network::IoSocketHandle>()) {}
```

Comment:
It'd be nice to avoid creating a unique pointer which will be overwritten with the socket() call.
TODO to clean this up in a follow-up?

Reply:
With this constructor, we won't call socket() I *think*.

Reply:
This constructor is only used in mock class which actually doesn't create a socket. I think passing in a unique_ptr is okay.